btree/btree.c

```diff
@@ -2815,6 +2815,31 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	 * Only the modified memory from the node needs to be updated.
 	 */
 }
+#define COPY_RECORD(tgt, src)                                                  \
+	({                                                                     \
+		struct m0_btree_rec *tgt_rec = (tgt);                          \
+		struct m0_btree_rec *src_rec = (src);                          \
+									       \
+		m0_bufvec_copy(&tgt_rec->r_key.k_data, &src_rec ->r_key.k_data,\
+			       m0_vec_count(&src_rec ->r_key.k_data.ov_vec));  \
+		m0_bufvec_copy(&tgt_rec->r_val, &src_rec->r_val,               \
+			       m0_vec_count(&src_rec ->r_val.ov_vec));         \
+	})
+
+#define COPY_VALUE(tgt, src)                                                   \
+	({                                                                     \
+		struct m0_btree_rec *tgt_rec = (tgt);                          \
+		struct m0_btree_rec *src_rec = (src);                          \
+									       \
+		m0_bufvec_copy(&tgt_rec->r_val, &src_rec->r_val,               \
+			       m0_vec_count(&src_rec ->r_val.ov_vec));         \
+	})
+
```

Comment:
Don't you need COPY_KEY macro ?

Reply:
Currently it is not needed. Therefore, I have not defined COPY_KEY macro